h2o-core/src/main/java/hex/ModelBuilder.java

```diff
@@ -234,6 +236,17 @@ private void setFinalState() {
     }
   }
 
+  private void saveModelCheckpointIfConfigured() {
+    Model model = _result.get();
+    if (model != null && !StringUtils.isNullOrEmpty(model._parms._export_checkpoints_dir)) {
+      try {
+        model.exportBinaryModel(model._parms._export_checkpoints_dir + "/" + model._key.toString(), true);
+      } catch (IOException e) {
+        throw new H2OIllegalArgumentException("export_checkpoints_dir", "saveModelIfConfigured", e);
```

Comment:
Hmm, I think this might be a costly mistake if the user inputs the directory incorrectly. This exception will stop CV models and prevent the main model from being built.

I think we should pre-create the target subdirectory for the model before it is actually built. This will make sure we have a write-access to the directory and that the directory exists.

I would also consider a special handling of the CV models - produce just a warning if we cannot write the CV model but fail if we cannot write the main model. But I am not 100% sure if this helps anything.

Reply:
good point. I think we should keep any advanced structuring of the checpoints dir to another task
as per not failing on CV models its would be the same case for main model, so we would only delay the exception as such I think its better to fail sonner rather than later

Reply:
sure, we can do the nested directories in another PR - definitely not necessary in this one

regarding the CV models - I think I agree - let's make it simpler (at least for now). My point of view is that the model checkpointing will most likely be used for long-running tasks (eg. overnight) - you don't want to find out that you ended up without a trained model just because you enabled checkpointing (that was supposed to prevent you from losing your work in the first place ;))